### PR TITLE
Refactor proxy logic in CommonServerPython

### PR DESCRIPTION
**Pull request by BenJoParadise**
## Status
Ready

## Description
Modified proxy logic functions in CommonServerPython:
enable_proxy is not needed (the logic is done in python loop)
disable_proxy changed to handle_proxy:
- checkbox logic implemented
- customizable proxy param name and checkbox default value
- returns proxies dictionary for `requests` methods "proxies" parameter.

Should check option for bypass mechanism that always disables proxy.

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
Describe additional changes done, for example adding a function to common server.

### CODE REVIEW

File: Scripts/script-CommonServerPython.yml [hunk 1]
```diff
@@ -28,31 +28,31 @@ script: |-
...
   dbotscores = {'Critical': 4, 'High': 3, 'Medium': 2,'Low': 1, 'Unknown': 0, 'Informational': 0.5}
 
-  SAVE_PROXIES = {}
-
-  def disable_proxy():
+  def handle_proxy(proxy_param_name='proxy', checkbox_default_value=False):
       """
-          Disable routing traffic through the system proxy.
+          Handle logic for routing traffic through the system proxy.
           Should usually be called at the beginning of the integration, depeding on proxy checkbox state.
```
Comment (yaakovi) [inline Scripts/script-CommonServerPython.yml]:
```suggestion
           Should usually be called at the beginning of the integration, depending on proxy checkbox state.
```

File: Scripts/script-CommonServerPython.yml [hunk 2]
```diff
@@ -28,31 +28,31 @@ script: |-
...
+      proxies = {}
+      if demisto.params().get(proxy_param_name, checkbox_default_value):
+          proxies = {
+              'http': os.environ.get('HTTP_PROXY') or os.environ.get('http_proxy', ''),
+              'https': os.environ.get('HTTPS_PROXY') or os.environ.get('https_proxy', '')
+          }
+      else:
+          for k in ('HTTP_PROXY', 'HTTPS_PROXY', 'http_proxy', 'https_proxy'):
+              if k in os.environ:
+                  del os.environ[k]
```
Comment (yaakovi) [inline Scripts/script-CommonServerPython.yml]: when will this be restored?

File: Integrations/integration-Centreon.yml [hunk 3]
```diff
@@ -36,8 +36,7 @@ script:
     # disable insecure warnings
     requests.packages.urllib3.disable_warnings()
 
-    if not demisto.params().get('proxy', False):
-        disable_proxy()
+    handle_proxy()
```
Comment (yaakovi) [inline Integrations/integration-Centreon.yml]: shouldn't we move this to the `EXECUTION CODE` section? 
i know until now we put it at the top, but is that really the best place?

also, this needs to be updated in the plugin template (after merging to master)

File: Scripts/script-CommonServerPython.yml (hunk 2 above)
Comment (glicht) [reply Scripts/script-CommonServerPython.yml]: @yaakovi the restore is done in the python loop. No need to perform this logic in common server.

File: Integrations/integration-Centreon.yml (hunk 3 above)
Comment (BenJoParadise) [reply Integrations/integration-Centreon.yml]: Theoretically yes.

However, if we do that, any request that will be sent before the `EXECUTION CODE` section will try to pass through proxy. I know it shouldn't happen, but i've already seen integrations with active code before that block.

Comment (BenJoParadise) [reply Integrations/integration-Centreon.yml]: In the case of this integration, yes, i'll do it now.

**Pull request merged**